packages/payment-processor/src/payment/single-request-proxy.ts

```diff
@@ -0,0 +1,187 @@
+  }
+
+  // Use artifact's default address for the payment chain
+  const singleRequestProxyFactory = singleRequestProxyFactoryArtifact.connect(
+    paymentChain as CurrencyTypes.EvmChainName,
+    signer,
+  );
+
+  if (!singleRequestProxyFactory.address) {
+    throw new Error(`SingleRequestProxyFactory not found on chain ${paymentChain}`);
+  }
+
+  const salt = requestPaymentNetwork?.values?.salt;
+  const feeAddress = requestPaymentNetwork?.values?.feeAddress;
+  const feeAmount = requestPaymentNetwork?.values?.feeAmount;
+  const paymentRecipient = requestPaymentNetwork?.values?.paymentAddress;
+
+  if (!salt || !feeAddress || !feeAmount || !paymentRecipient) {
+    throw new Error('Invalid payment network values');
+  }
+
+  const paymentReference = `0x${PaymentReferenceCalculator.calculate(
+    request.requestId,
+    salt,
+    paymentRecipient,
+  )}`;
+
+  const isERC20 =
+    requestPaymentNetwork.id === ExtensionTypes.PAYMENT_NETWORK_ID.ERC20_FEE_PROXY_CONTRACT;
+
+  let tx;
+
+  if (isERC20) {
+    const tokenAddress = request.currencyInfo.value;
+    tx = await singleRequestProxyFactory.createERC20SingleRequestProxy(
+      paymentRecipient,
+      tokenAddress,
+      paymentReference,
+      feeAddress,
+      feeAmount,
+    );
+  } else {
+    tx = await singleRequestProxyFactory.createEthereumSingleRequestProxy(
+      paymentRecipient,
+      paymentReference,
+      feeAddress,
+      feeAmount,
+    );
+  }
+
+  const receipt = await tx.wait();
+
+  const event = receipt.events?.[0];
+
+  if (!event) {
+    throw new Error('Single request proxy creation event not found');
+  }
+
+  const proxyAddress = ethers.utils.defaultAbiCoder.decode(['address', 'address'], event.data)[0];
+
+  if (!proxyAddress) {
+    throw new Error('Proxy address not found in event data');
+  }
+
+  return proxyAddress;
+}
+
+/**
+ * Executes a payment through a Single Request Proxy contract.
+ *
+ * @param singleRequestProxyAddress - The address of the deployed Single Request Proxy contract.
+ * @param signer - The Ethereum signer used to execute the payment transaction.
+ * @param amount - The amount to be paid, as a string representation of the value.
+ * @returns A Promise that resolves when the payment transaction is confirmed.
+ * @throws {Error} If the SingleRequestProxy contract is invalid.
+ * @throws {Error} If the proxy contract type cannot be determined, or if any transaction fails.
+ *
+ * @remarks
+ * This function supports both ERC20 and Ethereum payments.
+ * For ERC20 payments, it first transfers the tokens to the proxy contract and then triggers the payment.
+ * For Ethereum payments, it directly sends the Ether to the proxy contract.
+ * The function automatically detects whether the proxy is for ERC20 or Ethereum based on the contract interface.
+ */
+export async function payRequestWithSingleRequestProxy(
+  singleRequestProxyAddress: string,
+  signer: Signer,
+  amount: string,
+): Promise<void> {
```

Comment:
Is it possible to make this function (or another version of this function) take a `request` as a parameter instead of a `singleRequestProxyAddress`? This would align more closely with the other `pay<X>Request` functions like [payErc20Request()](https://github.com/RequestNetwork/requestNetwork/blob/44bf9fbde237839321aeb80955160c7eab8b0fb5/packages/payment-processor/src/payment/erc20.ts#L23-L67) and [payAnytoErc20ProxyRequest()](https://github.com/RequestNetwork/requestNetwork/blob/31cbead732308cf5bdad26c11a0c37f9c2face07/packages/payment-processor/src/payment/any-to-erc20-proxy.ts#L22-L49)

Reply:
It is possible but would come with the following :

- We'd have to query for the Single Request Proxy deployed with the same payment reference.
- There can be multiple single request proxies for one request since there is no exact limiter on them, this can cause ambiguity.
- A possibility to make payment through a wrong proxy that the user did not intend to pay through.

Reply:
Ah okay. Specifying a specific SRP address is better for now. 👍